filebeat/channel/util.go

```diff
@@ -60,7 +64,13 @@ func SubOutlet(out Outleter) Outleter {
 func (o *subOutlet) Close() error {
 	isOpen := o.isOpen.Swap(false)
 	if isOpen {
+		// Signal OnEvent() to terminate
 		close(o.done)
+		// This mutex prevents the event channel to be closed if OnEvent is
+		// still running.
+		o.mutex.Lock()
+		defer o.mutex.Unlock()
+		o.closeEventChannel()
```

Comment:
I am a bit confused , we have a call to `close(o.done)` this will trigger  the select below and call `o.closeEventChannel`

Reply:
The channel can still be close.

Reply:
OK, I got it,

OnEvent -> Acquire mutex
Out of bound close close -> Block on mutex if on event is currently running.

Reply:
@adriansr I am a bit worried that we one more synchronization mechanism to this object, we have the _isOpen_ (atomic bool), the `chClosed` has some execution state, the mutex and the channels.

I think I am missing some context from the environment and I am just trying to understand what we want to prevent, it is multiple calls to `close(o.ch)?

Reply:
```
// This mutex prevents the event channel to be closed if OnEvent is
// still running.
```

From your comment this is what we want to stop? So we could drop the atomic bool and use isOpen instead of `chClosed`

Reply:
by "drop" I mean using a single mutex.